crates/core/tests/fetch.rs

```diff
@@ -0,0 +1,278 @@
+        .unwrap();
+
+    // Wait until Alice has fetched Bob's and Bob has fetched Alice's ops.
+    tokio::time::timeout(Duration::from_millis(100), async {
+        loop {
+            tokio::time::sleep(Duration::from_millis(10)).await;
+            let ops_now_alice = op_store_alice
+                .retrieve_ops(requested_op_ids_bob.clone())
+                .await
+                .unwrap();
+            let ops_now_bob = op_store_bob
+                .retrieve_ops(requested_op_ids_alice.clone())
+                .await
+                .unwrap();
+            if ops_now_alice.len() == num_of_ops as usize
+                && ops_now_bob.len() == num_of_ops as usize
+            {
+                break;
+            }
+        }
+    })
+    .await
+    .unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn bob_comes_online_after_being_unresponsive() {
+    enable_tracing();
+    let fetch_config_alice = CoreFetchModConfig {
+        core_fetch: CoreFetchConfig {
+            first_back_off_interval_ms: 10,
+            re_insert_outgoing_request_delay_ms: 10,
+            ..Default::default()
+        },
+    };
+    let Peer {
+        fetch: fetch_alice,
+        op_store: op_store_alice,
+        transport: _transport_alice,
+        ..
+    } = make_peer(Some(fetch_config_alice), true).await;
+    let fetch_alice = fetch_alice.unwrap();
+
+    // Make Bob without fetch.
+    let Peer {
+        op_store: op_store_bob,
+        transport: transport_bob,
+        peer_url: peer_url_bob,
+        builder: builder_bob,
+        ..
+    } = make_peer(None, false).await;
+
+    let num_of_ops = 100;
+    let (requested_ops_bob, requested_op_ids_bob) = create_op_list(num_of_ops);
+
+    // Add Bob's ops to his store.
+    op_store_bob
+        .process_incoming_ops(
+            requested_ops_bob
+                .clone()
+                .into_iter()
+                .map(Into::into)
+                .collect(),
+        )
+        .await
+        .unwrap();
+
+    // Alice requests the ops from Bob.
+    fetch_alice
+        .request_ops(requested_op_ids_bob.clone(), peer_url_bob)
+        .await
+        .unwrap();
+
+    // Let some time pass for Alice to attempt sending a request to Bob
+    // and put him on the back off list.
+    tokio::time::sleep(Duration::from_millis(10)).await;
+
+    // Alice could not fetch ops from Bob.
+    let ops_in_store_alice = op_store_alice
+        .retrieve_ops(requested_op_ids_bob.clone())
+        .await
+        .unwrap();
+    assert_eq!(ops_in_store_alice, vec![]);
+
+    // Bob comes online. Fetch is kept to keep message handler live.
+    let _fetch_bob = builder_bob
+        .fetch
+        .create(
+            builder_bob.clone(),
+            TEST_SPACE_ID,
+            op_store_bob.clone(),
+            transport_bob,
+        )
+        .await
+        .unwrap();
+
+    // Wait until Alice has fetched Bob's ops.
+    tokio::time::timeout(Duration::from_millis(100), async {
```

Comment:
Worth extracting a helper for ops arrived? I did something similar in the gossip tests and maybe we could make a common one for this block of code based on a `DynOpStore` and a vector of `OpId`s

Reply:
Yes!